processor/resourcedetectionprocessor/internal/aws/eks/detector.go

```diff
@@ -74,9 +82,55 @@ func (d *detector) Detect(ctx context.Context) (resource pcommon.Resource, schem
 	d.rb.SetCloudProvider(conventions.AttributeCloudProviderAWS)
 	d.rb.SetCloudPlatform(conventions.AttributeCloudPlatformAWSEKS)
 
+	clusterName, err := d.getClusterName()
+	if err != nil {
+		return pcommon.NewResource(), "", err
+	}
+	d.rb.SetK8sClusterName(clusterName)
+
 	return d.rb.Emit(), conventions.SchemaURL, nil
 }
 
+func (d *detector) getClusterName() (string, error) {
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		d.logger.Debug("Failed to load default config", zap.Error(err))
```

Comment:
generally, we should log or return the error, not both.  I would expect whatever invokes Detect() to handle the error appropriately (e.g. by logging it).  Applies here and below.

Reply:
That's fair, I was mostly "inheriting" the style of the failure to detect an EKS environment in `Detect()`. In testing though, the messages are duplicated so I agree with your thought. I'll update.